lib/brook/test.ex

```diff
@@ -1,40 +1,31 @@
 defmodule Brook.Test do
-  def register() do
-    GenServer.cast(Brook.Driver.Test.via(), {:register, self()})
+  import Brook.Config, only: [registry: 1]
+  def register(instance) do
+    GenServer.cast(Brook.Driver.Test.via(instance), {:register, self()})
   end
 
-  @spec send(Brook.event_type(), Brook.author(), Brook.event()) :: :ok | {:error, Brook.reason()}
-  def send(type, author, data) do
-    register()
-    Brook.Event.send(type, author, data, %{module: Brook.Driver.Default, init_arg: []})
+  @spec send(Brook.instance(), Brook.event_type(), Brook.author(), Brook.event()) :: :ok | {:error, Brook.reason()}
+  def send(instance, type, author, data) do
+    register(instance)
+    Brook.Event.send(instance, type, author, data, %{module: Brook.Driver.Default, init_arg: []})
     :ok
   end
 
-  def save_view_state(event, collection, key, value) do
-    register()
-    storage = Brook.Config.storage()
-    apply(storage.module, :persist, [event, collection, key, value])
+  def with_event(instance, function) when is_function(function, 0) do
+    with_event(instance, fake_event(), function)
   end
 
-  def save_view_state(collection, key, value) do
-    save_view_state(fake_event(), collection, key, value)
+  def with_event(instance, event, function) when is_function(function, 0) do
```

Comment:
It feels like there should be a catch-all match clause that throws an error when function is not an actual function/ is wrong arity, instead of just throwing a bad match error.